library/modules/Maps.cpp

```diff
@@ -711,3 +712,398 @@ bool Maps::canStepBetween(df::coord pos1, df::coord pos2)
+*/
+
+/*
+This software is provided 'as-is', without any express or implied
+warranty.  In no event will the authors be held liable for any damages
+arising from the use of this software.
+Permission is granted to anyone to use this software for any purpose,
+including commercial applications, and to alter it and redistribute it
+freely, subject to the following restrictions:
+1. The origin of this software must not be misrepresented; you must not
+claim that you wrote the original software. If you use this software
+in a product, an acknowledgment in the product documentation would be
+appreciated but is not required.
+2. Altered source versions must be plainly marked as such, and must not be
+misrepresented as being the original software.
+3. This notice may not be removed or altered from any source distribution.
+*/
+
+namespace {
+    //----------------------------------------------------------------------------//
+    // Utility function
+    //
+    //----------------------------------------------------------------------------//
+    std::pair<bool, bool> check_tropicality(df::region_map_entry& region,
+        int y_pos
+    )
+    {
+        int flip_latitude = df::global::world->world_data->flip_latitude;
+
+        bool is_possible_tropical_area_by_latitude = false;
+        bool is_tropical_area_by_latitude = false;
+
+        if (flip_latitude == -1)  // NO POLES
+        {
+
+            // If there're no poles, tropical area is determined by temperature
+            is_possible_tropical_area_by_latitude = region.temperature >= 75;
+            is_tropical_area_by_latitude = region.temperature >= 85;
+
+        }
+
+        else
+        {
+            int v6;
+
+            df::world_data* wdata = df::global::world->world_data;
+
+            if (flip_latitude == 0) // NORTH POLE ONLY
+            {
+                v6 = y_pos;
+            }
+
+            else if (flip_latitude == 1) // SOUTH_POLE ONLY
+            {
+                v6 = df::global::world->world_data->world_height - y_pos - 1;
+            }
+
+            else if (flip_latitude == 2) // BOTH POLES
+            {
+                if (y_pos < wdata->world_height / 2)
+                    v6 = 2 * y_pos;
+                else
+                {
+                    v6 = wdata->world_height + 2 * (wdata->world_height / 2 - y_pos) - 1;
+                    if (v6 < 0)
+                        v6 = 0;
+                    if (v6 >= wdata->world_height)
+                        v6 = wdata->world_height - 1;
+                }
+
+            }
+
+            if (wdata->world_height == 17)
+                v6 *= 16;
+            else if (wdata->world_height == 33)
+                v6 *= 8;
+            else if (wdata->world_height == 65)
+                v6 *= 4;
+            else if (wdata->world_height == 129)
+                v6 *= 2;
+
+            is_possible_tropical_area_by_latitude = v6 > 170;
```

Comment:
```
/home/buildmaster/work/_E70955/_D0/src/library/modules/Maps.cpp:799:56: warning: 'v6' may be used uninitialized in this function [-Wmaybe-uninitialized]
```
Is "no poles" an option?

This looks like it was reverse-engineered, but it'd be nice to get rid of the warning anyway.

Reply:
It is possible to get a world with no poles. That possibility is covered by the prior code block; this code will never execute for a no-pole world.  But I'll force v6 to zero just to silence the warning.

And, yes, this is reverse-engineered code. I refactored it so it didn't look quite as fugly.